tests/jballerina-unit-test/src/test/resources/test-src/expressions/lambda/function-pointer-with-closure.bal

```diff
@@ -25,14 +25,16 @@ function testArrayFunctionInfer() returns string {
     return word;
 }
 
-function apply((function (any) returns any) f1, (function (any) returns any) f2) returns (function (any) returns any) {
-    return (arg) => f2(f1(arg));
+function apply((function (int) returns int) f1, (function (int) returns any) f2) returns (function (int) returns any) {
```

Comment:
Why do we have to change this? Can't we do the original logic anymore?

Reply:
According to the spec
`Function types are covariant in their return types and contravariant in the type of their parameter lists.`

Because of this to a function expecting any parameter, we cannot pass a function with a more specialized parameter. 

IE:
for a place where 
`function (any) returns any` is expected, we cannot pass `function (int) returns any`